test/unit/UpdatesPublisherTest.js

```diff
@@ -0,0 +1,114 @@
+require('require-self-ref')
+const test = require('ava')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
+const MockProducer = require('~/test/mocks/mockProducer')
+const MockChangesStream = require('~/test/mocks/mockChangesStream')
+const Promise = require('bluebird')
+proxyquire.noPreserveCache()
+
+function makeProducerAndStubs (sandbox) {
+  const createProducerStub = sandbox.stub()
+  const publisher = proxyquire('~/src/UpdatesPublisher', {
+    'windbreaker-service-util/queue': {
+      createProducer: createProducerStub
+    }
+  })
+  return {
+    createProducerStub: createProducerStub,
+    publisher: publisher
+  }
+}
+
+test.beforeEach('setup mock channel and connections', (t) => {
+  const producerOptions = {
+    queueName: 'test-queue'
+  }
+
+  const sandbox = sinon.sandbox.create()
+  const {publisher, createProducerStub} = makeProducerAndStubs(sandbox)
+
+  t.context = {
+    producerOptions,
+    sandbox,
+    publisher,
+    createProducerStub
+  }
+})
+
+test.afterEach('clean up', (t) => {
+  const { sandbox } = t.context
+
+  sandbox.restore()
```

Comment:
stopping the publisher relies on it's changes and producer properties to be defined, not always a given as I am testing logic paths rather than the whole shebang...will leave it to the test to stop publisher when necessary